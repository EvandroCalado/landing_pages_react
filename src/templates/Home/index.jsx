import { useEffect, useState } from 'react';
import Base from '../Base';
import { mapData } from '../../api/map-data';
import PageNotFound from '../PageNotFound';
import Loading from '../Loading';
import GridTwoColumn from '../../components/GridTwoColumn';
import GridContent from '../../components/GridContent';
import GridSection from '../../components/GridSection';
import GridImage from '../../components/GridImage';
import Pricing from '../../components/Pricing';
import SectionContact from '../../components/SectionContact';
import { useLocation } from 'react-router-dom';

const Home = () => {
  const [data, setData] = useState([]);
  const location = useLocation();

  useEffect(() => {
    const pathName = location.pathname.replace(/[^a-z0-9-_]/gi, '');
    const slug = pathName ? pathName : 'landing-page';

    const load = async () => {
      try {
        const data = await fetch(
          `http://localhost:1337/api/pages/?filters[slug]=${slug}&populate=deep`
        );
        const json = await data.json();
        const { attributes } = await json.data[0];
        const pageData = mapData([attributes]);

        setData(() => pageData[0]);
      } catch (error) {
        throw new Error(error);
      }
    };

    load();
  }, [location]);

  const { menu, sections, footerHtml, slug } = data;

  if (data === undefined) {
    return <PageNotFound />;
  }

  if (data && !data.slug) {
    return <Loading />;
  }

  return (
    <Base
      links={menu?.links}
      html={footerHtml}
      logoData={{ link: menu?.link, text: menu?.text, image: menu?.image }}
    >
      {sections.map((section, index) => {
        const { component } = section;
        const key = `${slug}-${index}`;

        if (component === 'section.section-two-columns') {
          return <GridTwoColumn key={key} {...section} />;
        }

        if (component === 'section.section-content') {
          return <GridContent key={key} {...section} />;
        }

        if (component === 'section.section-grid-text') {
          return <GridSection key={key} {...section} />;
        }

        if (component === 'section.section-grid-image') {
          return <GridImage key={key} {...section} />;
        }

        if (component === 'section.pricing') {
          return <Pricing key={key} {...section} />;
        }

        if (component === 'section.section-contact') {
          return <SectionContact key={key} {...section} />;
        }

        return null;
      })}
    </Base>
  );
};

export default Home;
