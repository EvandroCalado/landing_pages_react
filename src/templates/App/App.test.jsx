import Home from '.';
import { renderTheme } from '../../styles/render.theme';
import { screen } from '@testing-library/react';

test('renders learn react link', () => {
  const { debug } = renderTheme(<Home />);
  const headingContainer = screen.getByRole('heading', {
    name: 'Ol√°',
  }).parentElement;
  expect(headingContainer).toMatchSnapshot()
});
