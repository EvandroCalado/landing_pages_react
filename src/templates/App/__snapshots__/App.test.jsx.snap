// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  class="sc-bcXHqe hMwhAx"
>
  <h1>
    Ol√°
  </h1>
</div>
`;
